.github/workflows/provider-couchbase.yml

```diff
@@ -0,0 +1,150 @@
+name: provider-couchbase
+
+on:
+  push:
+    tags:
+      - 'provider-couchbase-v[0-9].[0-9]+.[0-9]+'
+  pull_request:
+    paths:
+      - '.github/workflows/provider-couchbase.yml'
+      - 'providers/couchbase/**'
+
+env:
+  working_directory: providers/couchbase
+permissions:
+  contents: read
+  packages: write
```

Comment:
Would you mind moving this into the `job` that requires the permission to help follow the [secure-repo recommendations](https://github.com/step-security/secure-repo?tab=readme-ov-file#1-automatically-set-minimum-github_token-permissions)?